tests/e2e/login/purge.spec.js

```diff
@@ -248,86 +247,76 @@ describe('Purging on login', () => {
 
   const restartSentinel = () => utils.stopSentinel().then(() => utils.startSentinel());
 
-  it('Logging in as a restricted user with configured purge rules should not download purged docs', () => {
-    utils.resetBrowser();
+  it('Logging in as a restricted user with configured purge rules should not download purged docs', async () => {
+    await utils.resetBrowser();
     commonElements.goToLoginPage();
     loginPage.login(restrictedUserName, restrictedPass);
     commonElements.calm();
     commonElements.goToReports();
     reports.expectReportsToExist([goodFormId]);
     reports.expectReportsToNotExist([badFormId]);
 
-    let purgeDate;
+    let result = await getPurgeLog();
+    // purge ran but after initial replication, nothing to purge
+    chai.expect(result._rev).to.equal('0-1');
+    chai.expect(result.roles).to.equal(JSON.stringify(restrictedUser.roles.sort()));
+    chai.expect(result.history.length).to.equal(1);
+    chai.expect(result.count).to.equal(0);
+    chai.expect(result.history[0]).to.deep.equal({
+      count: 0,
+      roles: result.roles,
+      date: result.date
+    });
+    const purgeDate = result.date;
+    
+    await utils.resetBrowser();
+    commonElements.calm();
+    await browser.waitForAngular();
+    result = await getPurgeLog();
+
+    // purge didn't run again on next refresh
+    chai.expect(result._rev).to.equal('0-1');
+    chai.expect(result.date).to.equal(purgeDate);
+
+    await browser.wait(() => utils.saveDocs(subsequentReports).then(() => true));
+    commonElements.sync();
+    commonElements.goToReports();
+    reports.expectReportsToExist([goodFormId, goodFormId2, badFormId2]);
 
-    return getPurgeLog()
-      .then(result => {
-        // purge ran but after initial replication, nothing to purge
-        chai.expect(result._rev).to.equal('0-1');
-        chai.expect(result.roles).to.equal(JSON.stringify(restrictedUser.roles.sort()));
-        chai.expect(result.history.length).to.equal(1);
-        chai.expect(result.count).to.equal(0);
-        chai.expect(result.history[0]).to.deep.equal({
-          count: 0,
-          roles: result.roles,
-          date: result.date
-        });
-        purgeDate = result.date;
-      })
-      .then(() => {
-        utils.resetBrowser();
-        commonElements.calm();
-        helper.waitForAngularComplete();
-        return getPurgeLog();
-      })
-      .then(result => {
-        // purge didn't run again on next refresh
-        chai.expect(result._rev).to.equal('0-1');
-        chai.expect(result.date).to.equal(purgeDate);
-      })
-      .then(() => {
-        browser.wait(() => utils.saveDocs(subsequentReports).then(() => true));
-        commonElements.sync();
-        commonElements.goToReports();
-        reports.expectReportsToExist([goodFormId, goodFormId2, badFormId2]);
+    await browser.wait(async() => {
```

Comment:
`browser.wait` is similar to the `await` command in that it waits for the promise to resolve before moving on. There are two options here I think. Revert to using the existing promise chaining and` browser.wait` or remove the promise chaining and switch it all to use `await`.